tests/Transport/ValueParserTest.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\PhpSnmp\Tests\Transport;
+
+use PHPUnit\Framework\TestCase;
+use SimPod\PhpSnmp\Exception\CannotParseUnknownValueType;
+use SimPod\PhpSnmp\Transport\ValueParser;
+
+final class ValueParserTest extends TestCase
+{
+    /**
+     * @param mixed $expected
+     *
+     * @dataProvider providerParse
+     */
+    public function testParse(string $raw, $expected) : void
+    {
+        self::assertSame($expected, ValueParser::parse($raw));
+    }
+
+    /** @return iterable<mixed> */
```

Comment:
```suggestion
    /** @return iterable<string, list<mixed>> */
```

Reply:
... testy ... je to úplně jedno :D

Reply:
jn ale jde lip leveragovat staticka analyza. treba ti to nekdy omylem zarve kdyz tam das blbost, se kterou jsi nepocital